File: java_stopwatch/Stopwatch.java
import java.util.Scanner;

public class Stopwatch {

    class Control {
        public final int refreshRateMsec = 1000;
        public final int counterRateMsec = 100;

        public final int countsPerSec = 1000 / counterRateMsec;
        public final int countsPerMin = 60 * countsPerSec;
        public final int countsPerHour = 3600 * countsPerSec;
        public volatile int count = 0;
        public volatile boolean run = true;
      }

    final Control control = new Control();

    class Counter
        implements Runnable {
        public void run() { 
            try {
                while(control.run){
                    Thread.sleep(control.counterRateMsec);
                    control.count+=1; // to be safe we might want to increase count from a separate, "synchronized void increment()" method, but same issue as in python...
                }
            } catch (InterruptedException e) {
                System.out.format("Counter thread was interrupted!");
            }
        }
    }

    class Refresh
        implements Runnable {
        public void run() {
            try {
                while(control.run){
                    //System.out.print("\033[H\033[2J");
                    //System.out.flush();
                    System.out.format(countToTime(control.count));
                    Thread.sleep(control.refreshRateMsec);
                }
            } catch (InterruptedException e) {
                System.out.format("Refresh thread was interrupted");
            }
        }
    }
    
    public String countToTime(int count){
        int hours = count / control.countsPerHour;
        int reminder = count % control.countsPerHour;
        int minutes = reminder / control.countsPerMin;
        int reminder2 = reminder % control.countsPerMin;
        int seconds = reminder2 / control.countsPerSec;
        int milisec = reminder2 % control.countsPerSec;
        
        return (hours + ":" + minutes + ":" + seconds + ":" + milisec + "\n");
    }

    public void readInput(){
        Scanner scanner = new Scanner(System.in);

        while (true){
            String keyInput = scanner.nextLine();
            if (keyInput.equals("q")){
                control.run = false;
                break;
            }
            else if (keyInput.equals("r")){
                control.count = 0;
            }
        }
        scanner.close();
    }

    public static void main(String[] args)
    throws InterruptedException {
        Stopwatch stopwatch = new Stopwatch();

        Thread counterThread = new Thread(stopwatch.new Counter());
        Thread refreshThread = new Thread(stopwatch.new Refresh());

        refreshThread.start();
        counterThread.start();

        stopwatch.readInput();

        refreshThread.join();
        counterThread.join();
    }
        
}

File: erlang_stopwatch/stopwatch.erl
-module(stopwatch).
-export([start/0]).
-define(REFRESH_RATE_MSEC, 1000).
-define(COUNTER_RATE_MSEC, 100).
-define(TIMEOUT_MSEC, (?COUNTER_RATE_MSEC * 100)).
-define(COUNTS_PER_SEC, (1000 div ?COUNTER_RATE_MSEC)).
-define(COUNTS_PER_MIN, (60 * ?COUNTS_PER_SEC)).
-define(COUNTS_PER_HOUR, (3600 * (?COUNTS_PER_SEC))).

start() ->
    Counter_PID = spawn(fun() -> counter(0) end),
    spawn(fun() -> refresh(Counter_PID) end),
    read_input(Counter_PID).

counter(Count) ->
    NewCount = receive
                    {tell_count, PID} -> 
                        PID ! {Count, self()},
                        Count;
                    {reset_count, PID} ->
                        PID ! {ok, self()},
                        0
                after
                    0 -> Count
                end,
    sleep_milisec(?COUNTER_RATE_MSEC),
    counter(NewCount+1).

sleep_milisec(Milisec) ->
    receive
        after Milisec -> continue
    end.

read_input(Counter_PID) ->
    case io:get_chars(">", 1) of
        "q" -> init:stop();
        "r" -> Counter_PID ! {reset_count, self()};
        _ -> ignore
    end,
    read_input(Counter_PID).

count_to_time(Count) ->
    Hours = Count div ?COUNTS_PER_HOUR,
    Reminder= Count rem ?COUNTS_PER_HOUR,
    Minutes = Reminder div ?COUNTS_PER_MIN,
    Reminder2 = Reminder rem ?COUNTS_PER_MIN,
    Seconds = Reminder2 div ?COUNTS_PER_SEC,
    Milisec = Reminder2 rem ?COUNTS_PER_SEC,
    lists:concat(lists:join(":",[Hours,Minutes,Seconds,Milisec])).

refresh(PID) ->
    PID ! {tell_count,self()},
    receive
        {Count, PID} -> 
            Time = count_to_time(Count),
            io:format("~s~n",[Time])
    after
        ?TIMEOUT_MSEC -> 
            io:fwrite("Timeout waiting for count from ~p.~n", [PID]),
            init:stop()
    end,
    sleep_milisec(?REFRESH_RATE_MSEC),
    refresh(PID).
